### PR TITLE
feat(run): Always discover network controller

### PR DESCRIPTION
**Pull request by LucaSeri**
### Prerequisite checklist

  - [x] Read the [contribution guidelines](https://unikraft.org/docs/contributing/) regarding submitting new changes to the project;
  - [x] Tested your changes against relevant architectures and platforms;
  - [x] Ran `make fmt` on your commit series before opening this PR;
  - [ ] Updated relevant documentation.

### Description of changes
This PR moves the network controller discovery from ``Pre`` to ``Run``, such that calling ``Run`` programmatically discovers the driver correctly.

### CODE REVIEW

File: internal/cli/kraft/run/run.go [hunk 1]
```diff
@@ -244,6 +219,35 @@ func (opts *RunOptions) discoverMachineController(ctx context.Context) error {
 	return nil
 }
 
+func (opts *RunOptions) discoverNetworkController(ctx context.Context) error {
+	var err error
+
+	if opts.Network == "" && opts.IP != "" {
+		return fmt.Errorf("cannot assign IP address without providing --network")
```
Comment (nderjung) [inline internal/cli/kraft/run/run.go:226]: This essentially prevents users from instances without a network. Instead, simply `return nil`.

File: internal/cli/kraft/run/run.go [hunk 2]
```diff
@@ -252,6 +256,11 @@ func (opts *RunOptions) Run(ctx context.Context, args []string) error {
 		return err
 	}
 
+	err = opts.discoverNetworkController(ctx)
+	if err != nil {
```
Comment (nderjung) [inline internal/cli/kraft/run/run.go]: You can inline this:
```go
if err := opts.discoverNetworkController(ctx); err != nil {
```

Comment (LucaSeri) [reply internal/cli/kraft/run/run.go]: Thanks! I inlined this one and the one above for consistency

**Review by nderjung — APPROVED** [review]
Awesome, thanks!

**Pull request merged**